competitions/MICCAI/surgtoolloc/classification_files/train.py

```diff
@@ -0,0 +1,235 @@
+            scaler=scaler,
+            writer=writer,
+            epoch=epoch,
+            iteration=i,
+            step=step,
+            loss_function=loss_function,
+        )
+
+        val_metric = run_eval(
+            model=model,
+            val_dataloader=val_dataloader,
+            cfg=cfg,
+            writer=writer,
+            epoch=epoch,
+            metric=metric,
+        )
+
+        if val_metric > best_metric:
+            print(f"SAVING CHECKPOINT: val_metric {best_metric:.5} -> {val_metric:.5}")
+            best_metric = val_metric
+
+            checkpoint = create_checkpoint(
+                model,
+                optimizer,
+                epoch,
+                scheduler=scheduler,
+                scaler=scaler,
+            )
+            torch.save(
+                checkpoint,
+                f"{cfg.output_dir}/fold{cfg.fold}/checkpoint_best_metric.pth",
+            )
+
+
+def run_train(
+    model,
+    train_dataloader,
+    optimizer,
+    scheduler,
+    cfg,
+    scaler,
+    writer,
+    epoch,
+    iteration,
+    step,
+    loss_function,
+):
+    model.train()
+    losses = []
+    progress_bar = tqdm(range(len(train_dataloader)))
+    tr_it = iter(train_dataloader)
+
+    for itr in progress_bar:
+        batch = next(tr_it)
+        inputs, labels = batch["input"].to(cfg.device), batch["label"].to(cfg.device)
+        iteration += 1
+
+        step += cfg.batch_size
+        torch.set_grad_enabled(True)
+        if torch.rand(1) > 0.5:
+            inputs, labels_a, labels_b, lam = mixup_data(inputs, labels)
+            with autocast():
+                outputs = model(inputs)
+                loss = lam * loss_function(outputs, labels_a) + (
+                    1 - lam
+                ) * loss_function(outputs, labels_b)
+        else:
+            with autocast():
+                outputs = model(inputs)
+                loss = loss_function(outputs, labels)
+        losses.append(loss.item())
+
+        scaler.scale(loss).backward()
+        torch.nn.utils.clip_grad_norm_(model.parameters(), 1.0)
+        scaler.step(optimizer)
+        scaler.update()
+        optimizer.zero_grad()
+        scheduler.step()
+
+        progress_bar.set_description(f"loss: {np.mean(losses):.2f}")
+
+
+def run_eval(model, val_dataloader, cfg, writer, epoch, metric):
+
+    model.eval()
+    torch.set_grad_enabled(False)
+
+    progress_bar = tqdm(range(len(val_dataloader)))
+    tr_it = iter(val_dataloader)
+
+    for itr in progress_bar:
+        batch = next(tr_it)
+        inputs, labels = batch["input"].to(cfg.device), batch["label"].to(cfg.device)
+        outputs = model(inputs)
+        outputs = (torch.sigmoid(outputs) > cfg.clf_threshold).float()
+        metric(outputs, labels)
+    score = metric.aggregate()[0]
+    print(score)
+    score = torch.mean(score).item()
+    metric.reset()
+    writer.add_scalar("F1", score, epoch)
+
+    return score
+
+
+if __name__ == "__main__":
```

Comment:
I think this is a good use case to demonstrate `monai.bundle.ConfgiParser` -- creating CLI entry points like this https://github.com/Project-MONAI/tutorials/blob/main/bundle/hybrid_programming/scripts/train_demo.py

and convert [the cfg.py files](https://github.com/Project-MONAI/tutorials/blob/bcd8dd38fb4fdcaac8c2792e49d71a75de45866d/competitions/MICCAI/surgtoolloc/classification_files/configs/cfg_efnb4.py) to yaml files, what do you think @yiheng-wang-nv @Nic-Ma?